packages/web-pkg/tests/unit/services/loadingService.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { LoadingService } from '../../../src/services/loadingService'
+
+describe('LoadingService', () => {
+  beforeEach(() => {
+    jest.useFakeTimers('modern')
+  })
+  it('adds a task and sets it inactive initially', () => {
+    const service = new LoadingService()
+    const action = new Promise((resolve) => {
+      resolve(true)
+    })
+
+    service.addTask(() => action)
+    expect(service.isLoading).toBeFalsy()
+  })
+  it('adds a task and sets it active after the debounce', () => {
+    const service = new LoadingService()
+    const action = new Promise((resolve) => {
+      resolve(true)
+    })
+
+    service.addTask(() => action)
+    jest.runAllTimers()
```

Comment:
Maybe add another check *before* the timers are run to check `isLoading` is falsy?

Reply:
I already have a separate test that covers exactly this scenario. I like to have that more explicitly :)